util/oidc/oidc.go

```diff
@@ -376,6 +376,7 @@ func (a *ClientApp) HandleCallback(w http.ResponseWriter, r *http.Request) {
 	idToken, err := a.provider.Verify(idTokenRAW, a.settings)
 	if err != nil {
 		log.Warnf("Failed to verify token: %s", err)
+		log.Infof("Client IP: %s", r.RemoteAddr)
```

Comment:
Let's have one log statement and/or add client ip as a structured field for the warn message. I find two logs of different kind from the same block confusing. However, we may also log ip address regardless in the outside block together with some other metadata.

Reply:
Understood. I recommend including the IP address in all warning messages, not just this one.